From aabe6844a8f5e6afcd4a21feff3a8046b4c9b0f9 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 10 Feb 2014 17:12:04 +0800
Subject: [PATCH 438/442] ENGR00298378 PXP_LIB: the pxp_conf should be
 initialized before passing to driver

The pxp_conf is allocated by malloc(), so before passing it to
PXP driver, it should be initialzed to avoid some field may has
random value.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 test/pxp_lib_test/pxp_test.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/pxp_lib_test/pxp_test.c b/test/pxp_lib_test/pxp_test.c
index 1fd5983..ad6b766 100644
--- a/test/pxp_lib_test/pxp_test.c
+++ b/test/pxp_lib_test/pxp_test.c
@@ -290,6 +290,7 @@ int pxp_test(void *arg)
 
 	/* Configure the channel */
 	pxp_conf = malloc(sizeof (*pxp_conf));
+	memset(pxp_conf, 0, sizeof(*pxp_conf));
 	proc_data = &pxp_conf->proc_data;
 
 	/* Initialize non-channel-specific PxP parameters */
-- 
1.8.0

